synapse/storage/databases/main/pusher.py

```diff
@@ -396,42 +396,57 @@ def _delete_pushers(txn) -> int:
     async def _remove_deleted_email_pushers(
         self, progress: dict, batch_size: int
     ) -> int:
-        """A background update that deletes all pushers for email addresses no longer
-        associated with a user.
+        """A background update that deletes all pushers for deleted email addresses.
+
+        In previous versions of synapse, when users deleted their email address, it didn't
+        also delete all the pushers for that email address. This background update removes
+        those to prevent unwanted emails
+
+        Args:
+            progress: dict used to store progress of this background update
+            batch_size: the maximum number of rows to retrieve in a single select query
+
+        Returns:
+            The number of deleted rows
         """
 
         last_pusher = progress.get("last_pusher", 0)
 
         def _delete_pushers(txn) -> int:
 
             sql = """
-                SELECT p.id, p.user_name, p.add_id, p.pushkey
+                SELECT p.id, p.user_name, p.app_id, p.pushkey
                 FROM pushers AS p
-                    JOIN user_threepids AS t ON t.user_id=p.user_name
-                WHERE p.app_id = 'm.email'
-                    AND t.medium = 'email'
-                    AND t.address = p.pushkey
+                    LEFT JOIN user_threepids AS t
+                        ON t.user_id=p.user_name
```

Comment:
```suggestion
                        ON t.user_id = p.user_name
```

just for consistency with the below.